### ISSUE TITLE
Rule Request: Vertical Parameter Alignment

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#1033]
When refactoring, it's often very easy to mess up the vertical alignment of parameters in function declarations and calls:

```swift
func validateFunction(_ file: File, kind: SwiftDeclarationKind,
                    dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
```

We could (easily?) have a correctable rule for this.

**Comment by shoumikhin** [comment]
Does this rule apply to closure argument args? Eg:

```
func openConversations(completion: @escaping (_ records: [Conversation]?,
                                              _ error: NSError?) -> Void) {
```

Produces:
warning: Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. (vertical_parameter_alignment)

**Comment by nicolasmiari-unext** [comment]
I'm not able to ignore this rule somehow. My config file is working (other rules like `line_length:` are observed); however this has no effect:

    disabled_rules:
      - vertical_parameter_alignment

Also, `class_delegate_protocol` and `weak_delegate` can't be disabled either?

**Comment by jpsim** [comment]
@shoumikhin no.

@nicolasmiari-unext can you please file a new issue with specific steps to reproduce? Something along the lines of https://github.com/realm/SwiftLint/issues/368#issuecomment-173676723

**Comment by nicolasmiari-unext** [comment]
There must be something wrong with my project; I created a new one (from single iOS/View Application) and can't reproduce... Will report back if I find something.
